docs/TRANSCRIPTION.md

```diff
@@ -0,0 +1,222 @@
+
+![RESTful API in Console](assets/restful-api.png)
+
+## Define an Endpoint
+We will have two endpoints: `api/transcription/start.json` and `api/transcription/stop.json`. Because our endpoints need input information, we will use a `POST` request to pass the input as the request body.
+
+We will check that these inputs are not empty. If they are, we will return a Bad Request response using our helper function from `utils/sendResponse.ts`. 
+
+Our endpoints will need the following information in the body:
+- `api/transcription/start.json` will need `channel`
+- `api/transcription/stop.json` will need `taskId` and `builderToken`
+
+If everything is executed properly, we send a successful response using our helper function with data we will fill in later.
+
+```ts
+import type { APIContext } from "astro";
+
+export async function POST({ request }: APIContext) {
+    const { channel } = await request.json()
+
+    if (!channel) {
+        return sendBadRequest("channel is required")
+    }
+
+    return sendSuccessfulResponse("<return data>")
+}
+```
+
+## Make Request Helper Functions
+The rest of this guide will focus on the implementation of Cloud Recording, which includes calling the Agora API. To simplify the code and have a uniform request structure, we will set up a helper function in `utils/makeRequest.ts`. 
+
+The `makeRequest` function defines all the headers, executes the request, and returns the response. 
+
+The inputs will need the `method` (we will use GET and POST), `url`, `body`, and the `credential`. 
+
+```ts
+export const makeRequest = async (method: string, url: string, credential: string, body?: string) => {
+    const headers = new Headers({
+        "Authorization": "basic " + credential,
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": `${method}, OPTIONS`,
+    });
+
+    const res = await fetch(url, {
+        method: method,
+        headers: headers,
+        body: body
+    })
+    if (!res.ok) {
+        console.log(await res.text())
+        throw new Error("Failed to make request")
+    }
+    return res
+}
+```
+
+Another helper function in `utils/generateCredential.ts` will generate this credential. It is a base-64 encoded credential using the Customer ID and Customer Secret.
+
+```ts
+export const generateCredential = () => {
+const credential = import.meta.env.CUSTOMER_ID + ":" + import.meta.env.CUSTOMER_SECRET
+
+const base64_credential = btoa(credential)
+return base64_credential
+}
+```
+
+## Generate Transcription Resource
+The first step in transcribing your video calls is generating a real-time transcription resource. Behind the scenes, this spins up a backend Agora service that can transcribe video calls. 
```

Comment:
> this spins up a backend Agora service

All Agora services are backend, do you mean a cloud function or cloud instance?

Reply:
I'm actually not sure what RTT is behind the hood. Does it matter for this guide to be more specific than "backend"?

Reply:
I would remove the word backend and just leave it as Agora service.